tempesta_fw/http_parser.c

```diff
@@ -448,7 +449,7 @@ parse_int_hex(unsigned char *data, size_t len, unsigned long *acc)
 	unsigned char *p;
 
 	for (p = data; p - data < len; ++p) {
-		if (unlikely(isspace(*p) || (*p == ';')))
+		if (unlikely(IS_CR_OR_LF(*p) || (*p == ';')))
 			return p - data;
```

Comment:
It seems the parser logic is broken. Assume we receive

         GET / HTTP/1.1\r\n
         Transfer-Encoding: chunked\r\n
         \r\n
         \r\n

`parse_int_hex()` returns 0 from the `if` statement when its called at `RGen_BodyChunk` state. `parser->_tmp.acc` stays zero, so we add TFW_STR_COMPLETE to `body` string. `RGen_BodyChunkExt` finds the last CRLF and moves to `RGen_EoL`. `RGen__EoL` finds that `crlf` and `body` are complete and returns TFW_PASS from the FSM. However, there is no mandatory `last-chunk`.